src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -126,112 +100,63 @@ public SubscriptionState OpenSubscription(SubscriptionConnection connection)
             return subscriptionState;
         }
 
-        public unsafe void AcknowledgeBatchProcessed(long id, long lastEtag)
+        public async Task AcknowledgeBatchProcessed(long id, long lastEtag)
         {
-            var transactionPersistentContext = new TransactionPersistentContext();
-            using (var tx = _environment.WriteTransaction(transactionPersistentContext))
-            {
-                TableValueReader config;
-                GetSubscriptionConfig(id, tx, out config);
+            var changeVectorForEtag = GetChangeVectorFromEtag(lastEtag);                       
 
-                var subscriptionId = Bits.SwapBytes((ulong)id);
+            var command = new AcknowledgeSubscriptionBatchCommand(_db.Name)
+            {                
+                ChangeVector = changeVectorForEtag,
+                NodeTag = _serverStore.NodeTag,
+                SubscriptionEtag = id
+            };
 
-                int oldCriteriaSize;
-                var now = SystemTime.UtcNow.Ticks;
-                var ptr = config.Read(SubscriptionSchema.SubscriptionTable.CriteriaIndex, out oldCriteriaSize);
+            var etag = await _serverStore.SendToLeaderAsync(command);
 
-                JsonOperationContext context;
-                using (_db.DocumentsStorage.ContextPool.AllocateOperationContext(out context))
-                {
-                    var copy = context.GetMemory(oldCriteriaSize);
-                    Memory.Copy(copy.Address, ptr, oldCriteriaSize);
-                    var table = tx.OpenTable(_subscriptionsSchema, SubscriptionSchema.SubsTree);
-                    TableValueBuilder tvb;
-                    using (table.Allocate(out tvb))
-                    {
-                        tvb.Add((byte*)&subscriptionId, sizeof(long));
-                        tvb.Add(copy.Address, oldCriteriaSize);
-                        tvb.Add((byte*)&lastEtag, sizeof(long));
-                        tvb.Add((byte*)&now, sizeof(long));
-                        TableValueReader existingSubscription;
-                        Slice subscriptionSlice;
-                        using (Slice.External(tx.Allocator, (byte*)&subscriptionId, sizeof(long), out subscriptionSlice))
-                        {
-                            if (table.ReadByKey(subscriptionSlice, out existingSubscription) == false)
-                                return;
-                        }
-                        table.Update(existingSubscription.Id, tvb);
-                    }
-                    tx.Commit();
-                }
-            }
+            await _db.WaitForIndexNotification(etag);            
         }
 
-        public unsafe void GetCriteriaAndEtag(long id, DocumentsOperationContext context, out SubscriptionCriteria criteria, out long startEtag)
+        public unsafe void GetCriteriaAndEtag(long id, DocumentsOperationContext context, out SubscriptionCriteria criteria, out ChangeVectorEntry[] startChangeVector)
```

Comment:
Change the method name to match what it is doing. Probably can do that as a tuple result value now